### ISSUE TITLE
LuaBackend still not working even after new version

### ISSUE THREAD

**Issue by atera93** [Sirius902/LuaBackend#18]
I recently switched from Epic Games to Steam version of KH. I tried to load Kingdom Hearts but it would crash. What should I do?

**Comment by HydroSulphide** [comment]
What were the exact steps you tried? What system are you using? Where is your scripts folder and how does your LuaBackend.toml look like?

**Comment by atera93** [comment]
I am using Steam for Kingdom Hearts and I am using LuaBackend. My system goes something like this;

[kh1]
scripts = [{ path = "scripts/kh1/", relative = true }]
exe = "KINGDOM HEARTS FINAL MIX.exe"
game_docs = "My Games/KINGDOM HEARTS HD 1.5+2.5 ReMIX"

[kh2]
scripts = [{ path = "scripts/kh2/", relative = true }]
exe = "KINGDOM HEARTS II FINAL MIX.exe"
game_docs = "My Games/KINGDOM HEARTS HD 1.5+2.5 ReMIX"

[bbs]
scripts = [{ path = "scripts/bbs/", relative = true }]
exe = "KINGDOM HEARTS Birth by Sleep FINAL MIX.exe"
game_docs = "My Games/KINGDOM HEARTS HD 1.5+2.5 ReMIX"

[recom]
scripts = [{ path = "scripts/recom/", relative = true }]
exe = "KINGDOM HEARTS Re_Chain of Memories.exe"
game_docs = "My Games/KINGDOM HEARTS HD 1.5+2.5 ReMIX"

**Comment by HydroSulphide** [comment]
This seems right, did you copy all 3 files to the directory, where your exe is located? And are you playing on Linux/Steam Deck or Windows? Are your scripts located in the directory specified by your LuaBackend.toml? Do your games even work without the LuaBackend hook? Maybe you need to verify your game files via steam.

**Comment by atera93** [comment]
I copied the files onto my D:/ Steam directory. My games work without the LuaBackend hook.

**Comment by HydroSulphide** [comment]
Could you post screenshots of your file manager (directory and path where you put your lua files and also the one where you put your script folder) and also one of the content of your LuaBackend.toml? It's weird that it crashes, but maybe you are just putting them in the wrong location.

**Comment by atera93** [comment]
Where the scripts are
![image](https://github.com/Sirius902/LuaBackend/assets/173875293/d4adb0f3-f4d5-4a4f-a39f-433ea5af6b82)

Where my game is located (Steam Version)
![image](https://github.com/Sirius902/LuaBackend/assets/173875293/a4a1cc2b-de3b-4526-a639-1d0039be55bc)

**Comment by HydroSulphide** [comment]
Yeah this should be right. Are there specific mods that crash the game at start? Or does it also crash with no mod at all? Maybe you are using an outdated mod? Could you try checking without mods and identify the mod that causes the crash? Because lua backend should not cause any crash with your setup.

**Comment by Djokaire** [comment]
Hey !
How are you ?

I have the same issue actually. The game can't run with DBGHELP.ddl. I have a entrypoint symgetsymfromaddr64 alert message when I try to launch the game. If I change DBGHELP to DINPUT8, the game launches but nothing happens, scripts won't work.
I have the steam version btw

**Comment by HydroSulphide** [comment]
This error sounds like a mod tries to access an address, that isn't valid. Same question to you: What mods are you using? Does it only happen with one specific mod? Was the mod you are using updated after the steam release?

You shouldn't rename it to dinput8.dll on windows. That's for Linux/Proton.

**Comment by atera93** [comment]
> Yeah this should be right. Are there specific mods that crash the game at start? Or does it also crash with no mod at all? Maybe you are using an outdated mod? Could you try checking without mods and identify the mod that causes the crash? Because lua backend should not cause any crash with your setup.

I removed all my mods except for some that were recently downloaded and it worked. I think its because of the software switch from EGS to Steam.



> Hey ! How are you ?
> 
> I have the same issue actually. The game can't run with DBGHELP.ddl. I have a entrypoint symgetsymfromaddr64 alert message when I try to launch the game. If I change DBGHELP to DINPUT8, the game launches but nothing happens, scripts won't work. I have the steam version btw

Try removing all of your mods first and reintroducing some that were downloaded recently.

**Comment by Djokaire** [comment]
Sadly, any script causes a crash. :/

**Comment by NezomiFloof** [comment]
Can confirm as well, all scripts crash KH1.

**Comment by cahudson94** [comment]
To be honest (as a dev on the mods) this is not an issue with the backend but the mods themselves. If you can open the F2 prompt window the dlls from here are working correctly. Direct the questions where you have picked up the mods, it is likely one of three issues either the mod has not been updated (slightly less likely at this point), you have an outdated version of the file, or you are on a yet unsupported platform (Steam Deck has known issues but is hard to do dev work for). For my share of work All PC ports of the lua mods I maintain have been updated.

TL;DR work with the maintainers of the mods if F2 is working as it is a problem on that end not on the backend side.

**Comment by Sirius902** [comment]
If this is still an issue try and follow the below steps. If that still doesn't work feel free to re-open the issue, closing for now.

> This is likely due to changes after the new versions of the games came out. Look [here](https://github.com/Sirius902/LuaBackend/issues/12#issuecomment-2599362779) if you need a guide on migrating from a pre-v1.8.0 version or using Steam instead of EGS.
